File: OpenCV/11_contornos_componentes/cpp/main.cpp
#include <opencv2/opencv.hpp>
#include <iostream>
int main(){ auto b=cv::imread("binary.png",cv::IMREAD_GRAYSCALE); if(b.empty()) return 1; std::vector<std::vector<cv::Point>> c; cv::findContours(b,c,cv::RETR_EXTERNAL,cv::CHAIN_APPROX_SIMPLE); std::cout<<"contornos "<<c.size()<<"\n"; for(size_t i=0;i<c.size()&&i<10;i++) std::cout<<i<<" area "<<cv::contourArea(c[i])<<" perimetro "<<cv::arcLength(c[i],true)<<"\n"; }

File: OpenCV/11_contornos_componentes/python/main.py
import cv2
b=cv2.imread("binary.png",cv2.IMREAD_GRAYSCALE); assert b is not None
contours,_=cv2.findContours(b,cv2.RETR_EXTERNAL,cv2.CHAIN_APPROX_SIMPLE); print("contornos",len(contours));
for i,c in enumerate(contours[:10]): print(i,"area",cv2.contourArea(c),"perimetro",cv2.arcLength(c,True),"M00",cv2.moments(c)["m00"])
